### PR TITLE
Feature adpater integration

### PR DESCRIPTION
**Pull request by Exelord**
# Practically Ember Custom Actions V2
### 1. New config API
**CHANGED:**
I changed config attributes to be more meaningful.
- `type` -> `method` - The default has changed from `PUT` -> `POST`
- `promiseType` -> `responseType`
- `params` -> `queryParams`

**ADDED:**
I added a feature #24 so from now you can define:
- `adapterOptions` can be specified in config which will be passed to the adapter.

**REMOVED:**
- `urlFor` has been removed in favor of customizable `urlForCustomAction` method in the adpater in case of `customAction`.

### 2. Action's Private Interface
The private interface of `action` has been a little bit adjusted and documented. It has been prepared for adapter support.

### 3. Redesigned `url-builder`
The `url-builder` utils completely changed his role. Now it is only responsible for joining paths of the URL and query params.

### 4. `customAction`
There is a new method available `customAction` which will allow using integration with adapters and serializers

### 5. Adapter support
- `urlForCustomAction` - you can now customize your request's URL in the adapter by overriding this method in case of using `customAction` instead of `modelAction` or `resourceAction`

## TODO
- [x] - Tests
- [x] - More descriptive documentation
- [ ] - Migration guide
- [x] - Add warning in case of passing payload other than `Object`
___
@KillaPL what do you think about it? Feedback, review is welcome as hell! :) This is just a draft. A lot of things is still to discuss.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,31 +154,74 @@ It's great for API with request data format restrictions
...
+#### urlForCustomAction
+```js
+export default JSONAPIAdapter.extend(AdapterMixin, {
+  urlForCustomAction(modelName, id, snapshot, requestType, query) {
+    if (requestType === 'myPublishAction') {
+      return 'https://my-custom-api.com/publish'
+    }
+    
+    return this._super(...arguments);
+    let url = this._buildURL(modelName, id);
```
Comment (KillaPL) [inline README.md]: Isn't that a dead code in here? I mean, the `return` above will return, and it won't get to that code.

File: README.md [hunk 2]
```diff
@@ -154,31 +154,74 @@ It's great for API with request data format restrictions
...
+export default JSONAPIAdapter.extend(AdapterMixin);
+```
+
+Then you can customize following methods:
+
+#### urlForCustomAction
+```js
+export default JSONAPIAdapter.extend(AdapterMixin, {
+  urlForCustomAction(modelName, id, snapshot, requestType, query) {
+    if (requestType === 'myPublishAction') {
```
Comment (KillaPL) [inline README.md]: Awesome! Exactly what I wanted to achieve!: )

**Comment by KillaPL** [comment]
So the plan here is to include the mixin, and then I will be able to do `urlForCustomAction`, `methodForCustomAction` and `headersForCustomAction` ? Do I get it right?

**Comment by Exelord** [comment]
For now there is only `urlForCustomAction`. But still the API of this function should be reworked :/ We have to make it as much similar as possible to original Ember Adapter pattern.

File: README.md (hunk 1 above)
Comment (Exelord) [reply README.md]: you right ;)

**Pull request merged**